src/Raven.Server/ServerWide/Maintenance/ClusterObserver.cs

```diff
@@ -701,36 +739,60 @@ private CompareExchangeTombstonesCleanupState GetMaxCompareExchangeTombstonesEta
                     if (hasReport == false)
+                    {
+                        ForTestingPurposesOnly().DiagnosticLogBuilder?.AppendLine($"Missing report for node '{nodeTag}' for database '{state.Name}'");
                         return CompareExchangeTombstonesCleanupState.InvalidDatabaseObservationState;
+                    }
 
+                    ForTestingPurposesOnly().DiagnosticLogBuilder?.AppendLine($"Node '{nodeTag}': LastClusterWideTransactionRaftIndex = {report.LastClusterWideTransactionRaftIndex}");
                     var clusterWideTransactionIndex = report.LastClusterWideTransactionRaftIndex;
                     if (maxEtag == -1 || clusterWideTransactionIndex < maxEtag)
+                    {
                         maxEtag = clusterWideTransactionIndex;
+                        ForTestingPurposesOnly().DiagnosticLogBuilder?.AppendLine($"Updated maxEtag to {maxEtag} based on node '{nodeTag}' LastClusterWideTransactionRaftIndex");
+                    }
 
                     foreach (var kvp in report.LastIndexStats)
                     {
                         var lastIndexedCompareExchangeReferenceTombstoneEtag = kvp.Value.LastIndexedCompareExchangeReferenceTombstoneEtag;
+                        ForTestingPurposesOnly().DiagnosticLogBuilder?.AppendLine($"Node '{nodeTag}', index '{kvp.Key}': LastIndexedCompareExchangeReferenceTombstoneEtag = {lastIndexedCompareExchangeReferenceTombstoneEtag}");
                         if (lastIndexedCompareExchangeReferenceTombstoneEtag == null)
+                        {
+                            ForTestingPurposesOnly().DiagnosticLogBuilder?.AppendLine($"{nameof(lastIndexedCompareExchangeReferenceTombstoneEtag)} is null, continuing...");
                             continue;
+                        }
 
                         if (maxEtag == -1 || lastIndexedCompareExchangeReferenceTombstoneEtag < maxEtag)
+                        {
                             maxEtag = lastIndexedCompareExchangeReferenceTombstoneEtag.Value;
+                            ForTestingPurposesOnly().DiagnosticLogBuilder?.AppendLine($"Updated maxEtag to {maxEtag} based on node '{nodeTag}', index '{kvp.Key}'");
+                        }
 
                         if (maxEtag == 0)
+                        {
+                            ForTestingPurposesOnly().DiagnosticLogBuilder?.AppendLine($"maxEtag reached 0 for node '{nodeTag}', index '{kvp.Key}', returning {CompareExchangeTombstonesCleanupState.NoMoreTombstones}.");
                             return CompareExchangeTombstonesCleanupState.NoMoreTombstones;
+                        }
                     }
                 }
             }
 
             if (maxEtag == 0)
+            {
+                ForTestingPurposesOnly().DiagnosticLogBuilder?.AppendLine($"Final {nameof(maxEtag)} is 0, returning {CompareExchangeTombstonesCleanupState.NoMoreTombstones}.");
```

Comment:
this is wrong, you are executing a method that will always enable you those logs. You need to use the field, not the method.

Reply:
Done